extensions/common/context.cc

```diff
@@ -295,13 +269,69 @@ flatbuffers::DetachedBuffer extractLocalNodeFlatBuffer() {
   return fbb.Release();
 }
 
+PeerNodeInfo::PeerNodeInfo(const std::string_view peer_metadata_id_key,
+                           const std::string_view peer_metadata_key) {
+  found_ = getValue({peer_metadata_id_key}, &peer_id_);
+  if (found_) {
+    getValue({peer_metadata_key}, &peer_node_);
+    return;
+  }
+
+  // Construct a fallback peer node metadata based on endpoint labels if it is
+  // not in filter state.
+  std::string endpoint_labels;
+  fallback_peer_node_ = extractEmptyNodeFlatBuffer();
+  if (!getValue(
+          {"upstream_host_metadata", "filter_metadata", "istio", "workload"},
+          &endpoint_labels)) {
+    return;
+  }
+  std::vector<std::string_view> parts = absl::StrSplit(endpoint_labels, ';');
+  // workload label should semicolon separated four parts string:
+  // workload_name|namespace|canonical_service|canonical_revision.
```

Comment:
Clearer to use workload_name;namespace; ... instead of |

Reply:
Perhaps create  ‘ flatbuffers::DetachedBuffer parseNodeFromString(std::string)’ ?

Reply:
Thanks for catching this! I think it is fine to keep it as is and do not extract it out, since the logic is not shared anywhere.